geowebcache/diskquota/jdbc/src/main/java/org/geowebcache/diskquota/jdbc/JDBCQuotaStoreFactory.java

```diff
@@ -271,6 +302,27 @@ private DataSource getH2DataSource(DefaultStorageFinder cacheDirFinder)
         return dataSource;
     }
 
+    /** Prepares a simple data source for the embedded HSQL */
+    private DataSource getHSQLDataSource(DefaultStorageFinder cacheDirFinder)
+            throws ConfigurationException {
+        File storeDirectory =
+                new File(cacheDirFinder.getDefaultPath(), "diskquota_page_store_hsql");
+        storeDirectory.mkdirs();
+
+        BasicDataSource dataSource = new BasicDataSource();
+
+        dataSource.setDriverClassName("org.hsqldb.jdbcDriver");
+        String database = new File(storeDirectory, "diskquota").getAbsolutePath();
+        dataSource.setUrl("jdbc:hsqldb:" + database + ";shutdown=true");
```

Comment:
There are a couple of issues with this URL.

The HSQL manual states, about that property, "This feature is useful for running tests, where it may not be practical to shutdown the database after each test. But it is not recommended for application programs."
If it's needed for tests, we might want to have a way to specify its presence with a system variable used only for tests.

The other issue is that, I believe, the database is currently kept in memory. Here is what I see in the database directory for disk quota:

![image](https://github.com/GeoWebCache/geowebcache/assets/325433/e7ef925f-9eae-4692-a99c-1aa1f3b7e2d8)

Compare with the HSQL BD we use for the GeoTools coordinate reference system database:

![image](https://github.com/GeoWebCache/geowebcache/assets/325433/7bfa7019-466e-4419-9719-1e066fa6d55f)

Checking [the documentation](http://hsqldb.org/doc/2.0/guide/running-chapt.html#rgc_inprocess), I believe we need to use "jdbc:hsqldb:file:..." for the database to be persisted.

Particularly important, is to check that you can abruptly kill the GWC process, and the database contents are not lost (because it can happen that the process dies without going through an orderly shutdown).

Reply:
all done